packages/compass-export-to-language/src/modules/transpiler.ts

```diff
@@ -18,11 +24,20 @@ type QueryExpression = {
   maxTimeMS?: string;
 };
 
-export type InputExpression = AggregationExpression | QueryExpression;
+export function isQuery(exportMode?: ExportMode) {
+  return exportMode === 'Query' || exportMode === 'Delete Query';
+}
+
+export type InputExpression = WithExportMode &
+  (AggregationExpression | QueryExpression);
```

Comment:
Can it really be an aggregation when export mode is passed? Looking at the changes in bson-transpilers this seems to be only used for delete / update queries or maybe I'm reading this code wrong 🤔

Reply:
I added the exportMode to be explicit on what type of MQL we want to export, instead of the current logic that was inferring it based on the structure of the InputExpression. Right now it doesn't use the exportMode nowhere else on the aggregation but at some point I would like to refactor it so it uses the exportMode instead of doing structure magic.

WDYT?

Reply:
Aggregation export to language is currently triggered from one place, query from two + your new use case. Is it big enough for a separate task? If yes, let's open a ticket and add a TODO comment with a ticket number that we are planning to clean this up, otherwise this adds code that doesn't do anything and produces a type that is not actually supported by the app

Reply:
I'll do the refactor then :D